src/model/chord.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{Pitch, Quality};
+use combine::{Parser, Stream};
+use std::fmt;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Chord {
+    pub pitch: Pitch,
+    pub quality: Quality,
+}
+
+impl Chord {
+    pub fn new(pitch: Pitch, quality: Quality) -> Self {
+        Self { pitch, quality }
+    }
+
+    pub fn parser<Input>() -> impl Parser<Input, Output = Self>
+    where
+        Input: Stream<Token = char>,
+    {
+        (Pitch::parser(), Quality::parser()).map(|(pitch, quality)| Chord::new(pitch, quality))
+    }
+}
+
+impl fmt::Display for Chord {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(&format!("{}{}", self.pitch, self.quality))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use strum::IntoEnumIterator;
+
+    #[test]
+    fn parser() {
+        let chords: Vec<_> = Pitch::all()
+            .into_iter()
+            .flat_map(|p| Quality::iter().map(move |q| Chord::new(p, q)))
+            .collect();
+
+        let parsed: Vec<_> = chords
+            .iter()
+            .map(|c| c.to_string())
+            .map(|string| Chord::parser::<&str>().parse(&string).unwrap().0)
+            .collect();
+
+        assert_eq!(parsed, chords);
```

Comment:
I'm taking this approach for most parser tests:
- test that string -> model -> string results in the same thing as the original